notification/twilio/voice_test.go

```diff
@@ -1,12 +1,393 @@
+				Params: url.Values{
+					"msgBody":      []string{b64enc.EncodeToString([]byte(fmt.Sprintf("%s with a status update for alert 'Widget is Broken'. Something is Wrong", prefix)))},
+					"msgPause":     []string{"66"},
+					"msgSubjectID": []string{"3"},
+				},
+			},
+		},
+		"Verification Notification": {
+			input: mockInput{
+				prefix: prefix,
+				msg: notification.Verification{
+					Dest: notification.Dest{
+						ID:    "1",
+						Type:  notification.DestTypeVoice,
+						Value: "+16125551234",
+					},
+					CallbackID: "2",
+					Code:       1234,
+				},
+			},
+			expected: &VoiceOptions{
+				ValidityPeriod: time.Second * 10,
+				CallType:       CallTypeVerify,
+				CallbackParams: url.Values{"msgID": []string{"2"}},
+				Params: url.Values{
+					"msgBody":      []string{b64enc.EncodeToString([]byte(fmt.Sprintf("%s with your 4-digit verification code. The code is: %s. Again, your 4-digit verification code is: %s.", prefix, spellNumber(1234), spellNumber(1234))))},
+					"msgPause":     []string{"52", "77"},
+					"msgSubjectID": []string{"-1"},
+				},
+			},
+		},
+		"Bad Type": {
+			input: mockInput{
+				prefix: prefix,
+				msg: notification.ScheduleOnCallUsers{
+					Dest: notification.Dest{
+						ID:    "1",
+						Type:  notification.DestTypeVoice,
+						Value: "+16125551234",
+					},
+					CallbackID:   "2",
+					ScheduleID:   "3",
+					ScheduleName: "4",
+					ScheduleURL:  "5",
+				},
+			},
+			expectedErr: errors.New("unhandled message type: notification.ScheduleOnCallUsers"),
+		},
+		"Missing prefix": {
+			input: mockInput{
+				msg: notification.Test{
+					Dest: notification.Dest{
+						ID:    "1",
+						Type:  notification.DestTypeVoice,
+						Value: "+16125551234",
+					},
+					CallbackID: "2",
+				},
+			},
+			expectedErr: errors.New("No prefix provided"),
+		},
+		"no input": {
+			input: mockInput{
+				prefix: prefix,
+			},
+			expectedErr: errors.New("unhandled message type: <nil>"),
+		},
+	}
+
+	for name, tc := range testCases {
+		t.Run(name, func(t *testing.T) {
+			// Arrange / Act
+			otps, err := buildMessage(tc.input.prefix, tc.input.msg)
+
+			// Assert
+			assert.Equal(t, tc.expected, otps)
+			if tc.expectedErr != nil || err != nil {
+				// have to do it this way since errors.Errorf will never match due to memory alocations.
+				assert.Equal(t, tc.expectedErr.Error(), err.Error())
+			}
+		})
+	}
+}
+
+func BenchmarkBuildMessage(b *testing.B) {
+	for i := 0; i < b.N; i++ {
+		_, _ = buildMessage(
+			fmt.Sprintf("%d", i),
+			notification.Test{
+				Dest: notification.Dest{
+					ID:    "1",
+					Type:  notification.DestTypeVoice,
+					Value: fmt.Sprintf("+1612555123%d", i),
+				},
+				CallbackID: "2",
+			})
+	}
+}
+
+func TestProcessSayBody(t *testing.T) {
+	type mockInput struct {
+		resp          *twiMLResponse
+		msgBody       string
+		msgPauseIndex []int
+	}
+
+	testCases := map[string]struct {
+		input    mockInput
+		expected *twiMLResponse
+	}{
```

Comment:
resp is empty for all these, which adds noise, and specifying the indexes manually seems fragile.